Source/OCMock/OCMMacroState.m

```diff
@@ -40,21 +41,31 @@ + (OCMStubRecorder *)endStubMacro
     OCMStubRecorder *recorder = [[(OCMStubRecorder *)[globalState recorder] retain] autorelease];
     BOOL didThrow = [globalState invocationDidThrow];
     [threadDictionary removeObjectForKey:OCMGlobalStateKey];
-	if(didThrow == NO && [recorder wasUsed] == NO)
-	{
-		[NSException raise:NSInternalInconsistencyException
-					format:@"Did not record an invocation in OCMStub/OCMExpect/OCMReject.\n"
-						   @"Possible causes are:\n"
-						   @"- The receiver is not a mock object.\n"
-						   @"- The selector conflicts with a selector implemented by OCMStubRecorder/OCMExpectationRecorder."];
-	}
+	  if(didThrow == NO && [recorder wasUsed] == NO)
+	  {
+      OCMLocation *location = recorder.ocm_location;
+        NSString *explanation = @"Did not record an invocation in OCMStub/OCMExpect/OCMReject.\n"
+                                @"Possible causes are:\n"
+                                @"- The receiver is not a mock object.\n"
```

Comment:
I would prefer that we would  assert on beginStubMacro/beginExpectMacro/beginRejectMacro that the received object is  member of OCMockObject / NSProxy  (and this way give a much more informative message and have a proper stack trace)

Reply:
Sorry tzvist, could you clarify which received object are you talking about? If it's what I think you are saying, we don't have that info in the `begin*Macro` method.

Reply:
if I understand the exception correctly, it says that the problem is cause by invalid `invocation`  object so why cant we explicitly check for these cases in the begin, why do we need to look for side effects such as wasUsed or didThrow and not check explicit in the begin macro?

Reply:
after looking at it some more I have misunderstand the way OCMock finds the class that is stubbed, apparently my proposal would not work.